rs/bitcoin/kyt/src/state.rs

```diff
@@ -45,10 +46,48 @@ pub enum FetchTxStatus {
 /// input address will be computed and filled in.
 #[derive(Clone, Debug)]
 pub struct FetchedTx {
-    pub tx: Transaction,
+    pub tx: TransactionInputOutput,
     pub input_addresses: Vec<Option<Address>>,
 }
 
+/// Instead of storing the full Transaction data, we only
+/// store relevant bits, including inputs (which are previous
+/// outputs) and output addresses.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransactionInputOutput {
+    pub inputs: Vec<PreviousOutput>,
+    pub outputs: Vec<Address>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PreviousOutput {
+    pub txid: Txid,
+    pub vout: u32,
+}
+
+impl TryFrom<Transaction> for TransactionInputOutput {
+    type Error = bitcoin::address::FromScriptError;
+
+    fn try_from(tx: Transaction) -> Result<Self, Self::Error> {
+        let inputs = tx
+            .input
+            .iter()
+            .map(|input| PreviousOutput {
+                txid: Txid::from(*(input.previous_output.txid.as_ref() as &[u8; 32])),
```

Comment:
nit: Intellij warns me of an unnecesary cast
```suggestion
                txid: Txid::from(*(input.previous_output.txid.as_ref())),
```

Reply:
Unfortunately it fails to compile. Had to reverse this

Reply:
ha sorry about that, it seems there are indeed multiple implementations of `AsRef`.